packages/ui/components/input-file/src/LionInputFile.js

```diff
@@ -0,0 +1,848 @@
+  ];
+
+  static get validationTypes() {
+    return ['error', 'info'];
+  }
+
+  /**
+   * @configure SlotMixin
+   */
+  // @ts-ignore
+  get slots() {
+    return {
+      ...super.slots,
+      input: () => {
+        // TODO: Find a better way to do value delegation via attr
+        const input = document.createElement('input');
+        const value = this.getAttribute('value');
+        if (value) {
+          input.setAttribute('value', value);
+        }
+        return input;
+      },
+      'file-upload-button': () => {
+        const button = document.createElement('button');
+        button.setAttribute('id', `upload-button-${this._inputId}`);
+        button.setAttribute('type', 'button');
+        button.setAttribute('click', `${() => this._inputNode.click()}`);
+        button.innerHTML = this.fileUploadButtonLabel;
+        return button;
+      },
+      'uploaded-file-list': () => ({
+        template: html`
+          <lion-uploaded-file-list
+            .fileList=${this._uploadedFilesMetaData}
+            .multiple=${this.multiple}
+          ></lion-uploaded-file-list>
+        `,
+        afterRender: this._enhanceUploadedList(),
+      }),
+    };
+  }
+
+  /**
+   * @type {HTMLInputElement}
+   * @protected
+   */
+  get _inputNode() {
+    return /** @type {HTMLInputElement} */ (super._inputNode);
+  }
+
+  get _fileUploadButtonNode() {
+    return this.querySelector(`#upload-button-${this._inputId}`);
+  }
+
+  /**
+   * The helpt text for the input node.
+   * When no light dom defined via [slot=help-text], this value will be used
+   * @type {string}
+   */
+  get fileUploadButtonLabel() {
+    return (
+      this.__fileUploadButtonLabel ||
+      (this._fileUploadButtonNode && this._fileUploadButtonNode.textContent) ||
+      ''
+    );
+  }
+
+  /**
+   * @param {string} newValue
+   */
+  set fileUploadButtonLabel(newValue) {
+    const oldValue = this.fileUploadButtonLabel;
+    /** @type {string} */
+    this.__fileUploadButtonLabel = newValue;
+    this.requestUpdate('fileUploadButtonLabel', oldValue);
+  }
+
+  /**
+   * @protected
+   * @configure FocusMixin
+   */
+  // @ts-ignore
+  get _focusableNode() {
+    return this._fileUploadButtonNode;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  get _isDragAndDropSupported() {
+    return 'draggable' in document.createElement('div');
+  }
+
+  constructor() {
+    super();
+    this.type = 'file';
+    /**
+     * @type {InputFile[]}
+     */
+    this._uploadedFilesMetaData = this._uploadedFilesMetaData || [];
+    /**
+     * @type {UploadResponse[]}
+     */
+    this.uploadResponse = [];
+    this.__initialUploadResponse = this.uploadResponse;
+    this.uploadOnFormSubmit = true;
+    this.multiple = false;
+    this.enableDragAndDrop = false;
+    this.isDragging = false;
+    this.maxFileSize = MAX_FILE_SIZE;
+    this.allowedFileTypes = '';
+    this.allowedFileExtensions = '';
+    this._isUserProvidedLabel = '';
```

Comment:
sounds more lik a boolean? Maybe `_hasUserProvidedFileUploadButtonLabel`? Or better: I think we should leave `fileUploadButtonLabel` to the Subclasser (who can provide it in a template, or, we provide a TemplateData object like in InputTelephone) like we do in all other components. 

The default text for Uploading a file ("Choose a file" by default in the platform) should be consistent across all forms within a Design System; an application developer should not change the button text for every instance. For a regular label/help-text, the latter does make sense